Gems/Multiplayer/Code/Include/Multiplayer/Components/NetworkHierarchyRootComponent.h

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Component/Component.h>
+#include <AzCore/Component/TransformBus.h>
+#include <Source/AutoGen/NetworkHierarchyRootComponent.AutoComponent.h>
+#include <Multiplayer/Components/NetworkHierarchyBus.h>
+
+namespace Multiplayer
+{
+    //! @class NetworkHierarchyRootComponent
+    //! @brief Component that declares the top level entity of a network hierarchy.
+    /*
+     * Call @GetHierarchicalEntities to get the list of hierarchical entities.
+     * A network hierarchy is meant to be a small group of entities. You can control the maximum supported size of
+     * a network hierarchy by modifying CVar @bg_hierarchyEntityMaxLimit.
+     *
+     * A root component marks either a top most root of a hierarchy, or an inner root of an attach hierarchy.
+     */
+    class NetworkHierarchyRootComponent final
+        : public NetworkHierarchyRootComponentBase
+        , public NetworkHierarchyRequestBus::Handler
+        , protected AZ::TransformNotificationBus::MultiHandler
```

Comment:
Isn't all of this exposed through AZ::Events already? I don't think we have any transform ebus touchpoints throughout any of the multiplayer gem or multiplayer sample code currently, it's all already been cut over to other more performant hooks.

Reply:
Oh, then that's a good call. Old habits die hard. I'll convert these over.

Reply:
Hmm, actually that will be a bit of a pain to do, since I'm using a ::MultiHandler, here. It means I'll have to create 2 events for each entity involved in the hierarchy and push them into a vector of AZ::Events. Hmm.

Reply:
This looks like too much work for a PR change in here. I'll create a follow up task for this as this change will not be straight forward.

Reply:
LYN-6601

Reply:
The CR has been converted over to use AZ::Events instead of ::MultiHandler.